spmd/compiler/fusion.py

```diff
@@ -32,25 +37,45 @@ class FusionElement:
     """This class tracks the nodes for a DTensor expanded communication collective in the graph"""
```

Comment:
right now it is DTensor specific in that the DTensor expansion will wrapper the all_reduce (or whatever comm) in a CommTensor, which adds the get_item and wait_comm nodes. 
I reuse the wait comm node and we search for it atm, so in that sense it is DTensor expansion specific.
It can be updated/adapted to any comm (from DTensor or not) but would need to add a check whether the signature wait_comm was present or not, and add it if needed.